services/clusters-keeper/src/simcore_service_clusters_keeper/modules/ec2.py

```diff
@@ -72,22 +74,27 @@ async def get_ec2_instance_capabilities(
         instance_type_names: set[InstanceTypeType],
     ) -> list[EC2InstanceType]:
         """instance_type_names must be a set of unique values"""
-        instance_types = await self.client.describe_instance_types(
-            InstanceTypes=list(instance_type_names)
-        )
-        list_instances: list[EC2InstanceType] = []
-        for instance in instance_types.get("InstanceTypes", []):
-            with contextlib.suppress(KeyError):
-                list_instances.append(
-                    EC2InstanceType(
-                        name=instance["InstanceType"],
-                        cpus=instance["VCpuInfo"]["DefaultVCpus"],
-                        ram=parse_obj_as(
-                            ByteSize, f"{instance['MemoryInfo']['SizeInMiB']}MiB"
-                        ),
+        try:
+            instance_types = await self.client.describe_instance_types(
+                InstanceTypes=list(instance_type_names)
+            )
+            list_instances: list[EC2InstanceType] = []
+            for instance in instance_types.get("InstanceTypes", []):
+                with contextlib.suppress(KeyError):
+                    list_instances.append(
+                        EC2InstanceType(
+                            name=instance["InstanceType"],
+                            cpus=instance["VCpuInfo"]["DefaultVCpus"],
+                            ram=parse_obj_as(
+                                ByteSize, f"{instance['MemoryInfo']['SizeInMiB']}MiB"
+                            ),
+                        )
                     )
-                )
-        return list_instances
+            return list_instances
+        except botocore.exceptions.ClientError as exc:
+            if exc.response.get("Error", {}).get("Code", "") == "InvalidInstanceType":
```

Comment:
We should have a place to classify `botocore` exceptions instead of using strings
For me moment, at least add a link to the definitions.

THOUGHT: if you map every single botocore error with your own, perhaps you can add the codes as a class attribute in your own exception, e.g.

```python
class Ec2InstanceTypeInvalidError(...)
    botocore_error_code = "InvalidInstanceType"
    
```
or the reverse mapping 
```python
botocore_to_ec2_exceptions = {"InvalidInstanceType": Ec2InstanceTypeInvalidError, ... }
```
or have the first (which is the dev perspective) and produce the last upon import.

Reply:
I am mapping only the ones I am interested in. but you are right I should raise a generic exception in the other case.